src/OrchardCore.Modules/OrchardCore.PublishLater/Migrations.cs

```diff
@@ -1,59 +1,84 @@
 using System;
 using OrchardCore.ContentManagement.Metadata;
 using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.ContentManagement.Records;
 using OrchardCore.Data.Migration;
 using OrchardCore.PublishLater.Indexes;
 using OrchardCore.PublishLater.Models;
 using YesSql.Sql;
 
-namespace OrchardCore.PublishLater
+namespace OrchardCore.PublishLater;
+
+public class Migrations : DataMigration
 {
-    public class Migrations : DataMigration
+    private readonly IContentDefinitionManager _contentDefinitionManager;
+
+    public Migrations(IContentDefinitionManager contentDefinitionManager)
     {
-        private readonly IContentDefinitionManager _contentDefinitionManager;
+        _contentDefinitionManager = contentDefinitionManager;
+    }
 
-        public Migrations(IContentDefinitionManager contentDefinitionManager)
-        {
-            _contentDefinitionManager = contentDefinitionManager;
-        }
+    public int Create()
+    {
+        _contentDefinitionManager.AlterPartDefinition(nameof(PublishLaterPart), builder => builder
+            .Attachable()
+            .WithDescription("Adds the ability to schedule content items to be published at a given future date and time."));
 
-        public int Create()
-        {
-            _contentDefinitionManager.AlterPartDefinition(nameof(PublishLaterPart), builder => builder
-                .Attachable()
-                .WithDescription("Adds the ability to schedule content items to be published at a given future date and time."));
-
-            SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
-                .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
-
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
-                    "DocumentId",
-                    nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
-
-            // Shortcut other migration steps on new content definition schemas.
-            return 2;
-        }
-
-        // This code can be removed in a later version.
-        public int UpdateFrom1()
-        {
-            // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
-            // this because dropping an index and altering a column don't work on all providers.
+        SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
+            .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
```

Comment:
You should update the Create method to include new column, so that a fresh setup defines all what is needed in one pass. And with Published then Latest, both at the end.

Reply:
I didn't touch the Create method for backward compatibility

Reply:
This is wrong, on an existing site/database the `Create()` method is not executed again.

Then on a fresh setup better to only execute an updated `Create()` method that returns the last version number, this in place of executing the old one and then migration steps that may add/remove things, particularly when migration steps may fail with some DB providers.